base/shell/cmd/dir.c

```diff
@@ -1019,7 +1024,15 @@ DirPrintWideList(PDIRFINDINFO ptrFiles[],       /* [IN] Files' Info */
             else
                 _stprintf(szTempFname, _T("%s"), ptrFiles[temp]->stFindInfo.cFileName);
 
-            DirPrintf(lpFlags, _T("%-*s"), iLongestName + 1, szTempFname);
+            if (bCJK)
+            {
+                cxWidth = ConGetTextWidth(szTempFname);
+                DirPrintf(lpFlags, _T("%s%*s"), szTempFname, iLongestName + 1 - cxWidth, _T(""));
```

Comment:
optional suggestion:
```suggestion
                /* Print string and add space padding */
                DirPrintf(lpFlags, _T("%s%*s"), szTempFname, iLongestName + 1 - cxWidth, _T(""));
```